### PR TITLE
added vectorize init arg to ExecComp

### PR DESCRIPTION
**Pull request by naylor-b**
- this change was done to support later PR that uses declared partials sparsity for coloring.
- changed default complex stepsize to 1e-40
- changed default variable value from 0.0 to 1.0

### CODE REVIEW

File: openmdao/components/exec_comp.py [hunk 1]
```diff
@@ -131,7 +140,7 @@ def __init__(self, exprs, **kwargs):
         -----
         If a variable has an initial value that is anything other than 0.0,
```
Comment (Kenneth-T-Moore) [inline openmdao/components/exec_comp.py]: Should this be reworded with the initial value change? "If a variable has an initial value that is anything other than  0.0 ..."

File: openmdao/test_suite/components/matmultcomp.py [hunk 2]
```diff
@@ -45,11 +45,6 @@ def compute(self, inputs, outputs):
         time.sleep(self.sleep_time)
```
Comment (Kenneth-T-Moore) [inline openmdao/test_suite/components/matmultcomp.py]: Do we even need the MatMultComp anymore?

File: openmdao/components/exec_comp.py [hunk 3]
```diff
@@ -342,16 +367,33 @@ def compute_partials(self, inputs, partials):
...
+                    pwrap[param][idx] += step
+
+                    uwrap = _TmpDict(self._outputs, return_complex=True)
+
+                    # solve with complex param value
+                    self._residuals.set_const(0.0)
+                    self.compute(pwrap, uwrap)
+
+                    for u in out_names:
+                        jval = imag(uwrap[u] / self.complex_stepsize)
```
Comment (Kenneth-T-Moore) [inline openmdao/components/exec_comp.py]: If uwrap[u] is an ndarry, then multiplying by the scalar (1.0 / complex_stepsize) is way faster than division.

File: openmdao/components/exec_comp.py [hunk 4]
```diff
@@ -342,16 +367,33 @@ def compute_partials(self, inputs, partials):
...
+                        if (u, param) not in partials:  # create the dict entry
+                            partials[(u, param)] = np.zeros((jval.size, psize))
```
Comment (Kenneth-T-Moore) [inline openmdao/components/exec_comp.py]: Whoa, we never have to create a dictionary entry in the jacobian inside of a component's `compute_partials`. Why do we have to do it in execcomp? (also, does coverage confirm that this gets called?  [both branches of if]

File: openmdao/test_suite/components/matmultcomp.py (hunk 2 above)
Comment (naylor-b) [reply openmdao/test_suite/components/matmultcomp.py]: Yeah it's still used for parallel FD testing.

File: openmdao/components/exec_comp.py (hunk 4 above)
Comment (naylor-b) [reply openmdao/components/exec_comp.py]: fixed

File: openmdao/components/exec_comp.py (hunk 3 above)
Comment (naylor-b) [reply openmdao/components/exec_comp.py]: fixed

File: openmdao/components/exec_comp.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/components/exec_comp.py]: fixed

**Pull request merged**